stix2/utils.py

```diff
@@ -284,7 +284,12 @@ def new_version(data, **kwargs):
 
     cls = type(data)
     if 'modified' not in kwargs:
-        kwargs['modified'] = get_timestamp()
+        old_modified = parse_into_datetime(data["modified"])
+        new_modified = get_timestamp()
+        # Ensure the new is newer than the old!
+        if new_modified <= old_modified:
+            new_modified = old_modified + dt.timedelta(microseconds=1000)
```

Comment:
Is there a reason to use `microseconds=1000` instead of `milliseconds=1`?

Reply:
`timedelta` doesn't have a milliseconds field, but I guess it does support it as a kwarg.  I hadn't realized that.

One thing I did think of though, is that if `new_modified` is later but less than a millisecond later, you may still wind up with a duplicate `modified` field due to loss of precision when it is truncated to milliseconds.  What is really needed is to ensure that `new_modified` and `old_modified` represent different milliseconds.  This is guaranteed if they're >= 1 millisecond different, but not necessarily if they are < 1 millisecond different (but it's still possible).  So it probably actually needs further modification.